client/tsconfig.json

```diff
@@ -0,0 +1,18 @@
+{
+  "include": ["./i18n/**/*", "./plugins/**/*","./src/**/*","./utils/**/*"],
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "esnext",
+    "moduleResolution": "node",
+    "lib": ["dom", "es2017"],
```

Comment:
Why `es2017`? Why not later? Is it a case of _only once we use `es2018+` features, we add it_ ?

Reply:
Good question - honest answer is that it was the setting in the Gatsby example project! However, we might be better off using es2020 for lib and target and relying on Babel to handle browser support.

@nhcarrigan does that sound right?

Reply:
es2020 for everything, probably. I.e. the latest version, but a specific version.

Reply:
So, I haven't really used TS with Babel (Angular supports TS out of the box) - but looking at this PR we aren't using the TypeScript compiler we're specifically using Babel. 

Additionally, the `lib` setting is used for libraries we want to include in the compilation, where the `target` setting is used to specify HOW we want to compile. I don't believe we need `es` anything in the `lib`, because our target is `esnext`.

Reply:
Yeah, good point about the parser.

 Isn't lib for setting the type declaration for that version of JS? So, if we set target to es2020 and remove lib entirely we should be good.

Reply:
Type defs, yes - that's what I meant.

I do think we need the `"dom"` lib, or we will get type errors on things like `window`.

Reply:
I did a bit more digging.  Turns out if you drop `lib` TS will rely on `target` for the JS declarations and include `"dom"` by default.  Since we want basically every declaration we can get, I think this works.